ConfirmAccount/business/AccountRequestSubmission.php

```diff
@@ -90,10 +90,13 @@ public function submit( IContextSource $context ) {
 		}
 		
 		//before we continue, verify user
-		$code = sha1($_SERVER['REMOTE_ADDR'] . date('m'));
-		$data = file_get_contents('http://scratch.mit.edu/site-api/comments/project/10135908/?page=1&salt=' . md5(time())); //add the salt so it doesn't cache
+		//the project link is stored in the interface, so splice the URL out of it (it should be the only decimal there)
+		$project_link = $context->msg('requestaccount-project-link')->text();
+		preg_match('%(\d+)%', $project_link, $matches);
+		$code = $context->getRequest()->getSessionData('confirmaccount-code');
```

Comment:
This allows storing the link to the verification project in the database if anybody wants to use a different one.